nova_vm/src/ecmascript/types/language/bigint.rs

```diff
@@ -11,3 +15,23 @@ pub enum BigInt {
     BigInt(BigIntIndex) = BIGINT_DISCRIMINANT,
     SmallBigInt(SmallInteger) = SMALL_BIGINT_DISCRIMINANT,
 }
+
+impl BigInt {
+    /// ### [6.1.6.2.12 BigInt::lessThan ( x, y )](https://tc39.es/ecma262/#sec-numeric-types-bigint-lessThan)
+    ///
+    /// The abstract operation BigInt::lessThan takes arguments x (a BigInt) and
+    /// y (a BigInt) and returns a Boolean. It performs the following steps when
+    /// called:
+    pub(crate) fn less_than(self, agent: &mut Agent, y: BigInt) -> bool {
```

Comment:
**suggestion:** I haven't really been following this pattern myself as of yet, but I've been thinking that it might be a good idea to actually move all abstract operations into their own modules, so eg. `types/language/bigint/abstract_operations.rs` in this case.

That way we have kind of a closer control over the API of our `Value` "subclasses" and it replicates the spec to a T.

Reply:
Interesting, what do you propose in the case of naming collisions? e.g. `Number::lessThan` vs abstract `LessThan`?

Reply:
Module scoping, so `number::abstract_operations::less_than` (or perhaps rather `number::less_than`) vs general `ecmascript::abstract_operations::less_than`. If both are needed in the same file then they'd need to be scoped, but I'm not sure if that's particularly common.

Reply:
But it's also fair to say that `Number::lessThan` in the spec could just as well be a method on `Number` enum.